lib/hypertrack.dart

```diff
@@ -53,49 +74,88 @@ class HyperTrack {
   /// In case of frequent changes, it is possible, that intermediate states
   /// could be lost, so if more real-time responsiveness is required it is
   /// recommended to use [addGeotag] for passing that data.
-  void setDeviceMetadata(Map<String, Object> data) =>
-      _pluginInterface.setDeviceMetadata(data);
+  Future<void> setMetadata(JSONObject data) {
+    return invokeSdkVoidMethod(
+        SdkMethod.setMetadata, serializeMetadata(data));
+  }
 
-  /// Enable debugging.
-  void enableDebugLogging() async {
-    const MethodChannel('sdk.hypertrack.com/handle')
-        .invokeMethod('enableDebugLogging');
+  Future<Result<Location, HyperTrackError>> addGeotag(JSONObject data) {
```

Comment:
There is a difference between HyperTrackError and LocationError. LocationError has unique starting and notTracking values which HyperTrackError shouldn't have. I made this change in iOS interface but forgot to update sdk-interface, will update that. Take a look [here](https://github.com/hypertrack/mobile-common/blob/891539ee43c12ae11f5dc50a026b9bf39cfa28d7/iOS/SDK/HyperTrack/HyperTrack.swift#L188-L249)

Reply:
In this case HyperTrackError is a transitional datatype that contains the sum of old and new errors. I thought once the new API be ready we will have another API change where wrapper APIs will be implemented 1-to-1 to the SDK API. I saw it in the repo but omited this change for the sake of simplicity. Does it make sense to separate LocationError in current stage?